##### 1. Write a Python Program to Find the Factorial of a Number?

In [1]:
num = int(input("Enter a number: "))

factorial = 1
for i in range(1, num + 1):
    factorial *= i

print("The factorial of", num, "is", factorial)


Enter a number: 6
The factorial of 6 is 720


##### 2. Write a Python Program to Display the multiplication Table?

In [4]:
# take input from the user
num = int(input("Enter a number: "))

# display the multiplication table
for i in range(1, 11):
   print(num, "x", i, "=", num*i)

Enter a number: 17
17 x 1 = 17
17 x 2 = 34
17 x 3 = 51
17 x 4 = 68
17 x 5 = 85
17 x 6 = 102
17 x 7 = 119
17 x 8 = 136
17 x 9 = 153
17 x 10 = 170


##### 3. Write a Python Program to Print the Fibonacci sequence?

The Fibonacci sequence is a series of numbers in which each number after the first two is the sum of the two preceding ones. The sequence goes like this: 0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584, 4181, 6765, 10946, 17711, 28657, 46368, 75025, 121393, 196418, 317811, 514229, 832040, 1346269, 2178309, 3524578, 5702887, 9227465, 14930352, 24157817, 39088169, 63245986, 102334155, 165580141, 267914296, 433494437, 701408733, 1134903170, 1836311903, etc.

In [5]:
# take input from the user
terms = int(input("Enter the number of terms: "))

# initialize variables
num1, num2 = 0, 1
count = 0

# check if the number of terms is valid
if terms <= 0:
   print("Please enter a positive integer.")
elif terms == 1:
   print("Fibonacci sequence upto",terms,":")
   print(num1)
else:
   print("Fibonacci sequence:")
   while count < terms:
       print(num1)
       nth = num1 + num2
       num1 = num2
       num2 = nth
       count += 1


Enter the number of terms: 5
Fibonacci sequence:
0
1
1
2
3


##### 4. Write a Python Program to Check Armstrong Number?

>An Armstrong number is a number that is equal to the sum of its own digits raised to the power of the number of digits. 

For example, 153 is an Armstrong number because it is a 3-digit number and the sum of the cubes of its digits is equal to 153, i.e., 1^3 + 5^3 + 3^3 = 153. Similarly, 371 is also an Armstrong number because 3^3 + 7^3 + 1^3 = 371.

In [6]:
# take input from the user
num = int(input("Enter a number: "))

# initialize variables
sum = 0
order = len(str(num))

# calculate sum of the cubes of each digit
temp = num
while temp > 0:
   digit = temp % 10
   sum += digit ** order
   temp //= 10

# display the result
if num == sum:
   print(num,"is an Armstrong number")
else:
   print(num,"is not an Armstrong number")


Enter a number: 4633
4633 is not an Armstrong number


In this program, we first take an integer input from the user using the input() function and int() function. We store this input in the variable num.

Then, we initialize a variable sum to 0 and calculate the order of the number, which is the number of digits in the number, using the len() function and str() function.

We then use a while loop to calculate the sum of the cubes of each digit in the number. In each iteration of the loop, we extract the last digit of the number using the modulo operator %, calculate its cube, and add it to the sum. We then remove the last digit from the number using integer division //. We repeat this process until all the digits of the number have been processed.

Finally, we use an if statement to check if the sum of the cubes of the digits is equal to the original number. If it is, we print a message indicating that the number is an Armstrong number. Otherwise, we print a message indicating that the number is not an Armstrong number.

##### 5. Write a Python Program to Find Armstrong Number in an Interval?

In [ ]:
# define a function to check if a number is an Armstrong number
def is_armstrong(num):
    # convert the number to a string to count the digits
    num_str = str(num)
    # count the number of digits in the number
    num_digits = len(num_str)
    # calculate the sum of the digits raised to the power of the number of digits
    digit_sum = sum([int(digit) ** num_digits for digit in num_str])
    # if the sum of the digits raised to the power of the number of digits is equal to the number, it is an Armstrong number
    if digit_sum == num:
        return True
    else:
        return False

# take user input for the lower and upper limits of the interval
lower = int(input("Enter lower range: "))
upper = int(input("Enter upper range: "))

# iterate over the range of numbers and check if they are Armstrong numbers
for num in range(lower, upper + 1):
    if is_armstrong(num):
        print(num)


##### 6. Write a Python Program to Find the Sum of Natural Numbers?

In [7]:
# take input from the user
num = int(input("Enter a positive integer: "))

# initialize variables
sum = 0

# use a loop to add up the numbers
for i in range(1, num+1):
    sum += i

# display the result
print("The sum of the first", num, "natural numbers is:", sum)


Enter a positive integer: 10
The sum of the first 10 natural numbers is: 55
